rosidl_typesupport_connext_c/cmake/rosidl_typesupport_connext_c_generate_interfaces.cmake

```diff
@@ -164,11 +166,15 @@ if(WIN32)
   target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
     PRIVATE "ROSIDL_TYPESUPPORT_CONNEXT_C_BUILDING_DLL")
   target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+    PRIVATE "ROSIDL_GENERATOR_C_${PROJECT_NAME}_BUILDING_DLL")
+  target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
     PRIVATE "NDDS_USER_DLL_EXPORT_${PROJECT_NAME}")
 endif()
 # The following still uses CPP because the Connext code which uses it was generated for CPP.
 target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
   PRIVATE "ROSIDL_TYPESUPPORT_CONNEXT_CPP_BUILDING_DLL_${PROJECT_NAME}")
+target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+  PRIVATE "ROSIDL_GENERATOR_C_BUILDING_DLL_${PROJECT_NAME}")
```

Comment:
This library defines a function which is forward-declared in a header provided by `rosidl_generator_c` with `ROSIDL_GENERATOR_C_PUBLIC_@(pkg_name)`:

https://github.com/ros2/rosidl/blob/master/rosidl_generator_c/resource/msg__type_support.h.em#L46-L49

I'm following the suggestion of @wjwwood, who designed the `ROSIDL_GET_TYPE_SUPPORT` mechanism. Maybe he can weigh in here.